### ISSUE TITLE
Provide method to fetch FlashMessages from Request

### ISSUE THREAD

**Issue by rieschl** [mezzio/mezzio-flash#2]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | no

#### Summary

In my own middlewares I like to add a static method in the Middleware to fetch the respective instance from the reqeust. This way I don't have to check everytime if the return is `null` and also don't have to manually assign the type for my IDE. For the `FlashMessageMiddleware` that would look something like that:
```
    public static function flashMessages(ServerRequestInterface $request): FlashMessages
    {
        return $request->getAttribute(self::FLASH_ATTRIBUTE);
    }
```
If the middleware has not been processed in the pipeline, a `TypeError` is thrown, so I can be sure I'll always get the correct object.
The problem with the current implementation of the `FlashMessageMiddleware` is, that you can set a custom `attributeKey`, so that complicates it a little bit (in my middlewares the attribute constants are private, so the only "official" way to fetch the object from the Request is via the static method). That could be mitigated in providing an optional second (deprecated?) parameter for the static method. Why did you event allow setting a custom attribute key?
Would you be open for such a change or should I keep using my own wrapper for this middleware?

**Comment by weierophinney** [comment]
> Why did you event allow setting a custom attribute key?

To allow usage of multiple different `FlashMessageMiddleware` instances that were capable of segregating their own sets of messages, without having to worry about naming collisions. When such a scenario is in play, you either hard-code in the other middleware and handlers the other attribute value, or, better, _inject it as a dependency_.

We tend to discourage static methods within the framework, as they lead to tight coupling, and discourage the ability to have multiple configurations — which is exactly what the configurable attribute key is for.

One possible alternative would be to provide an instance method on the `FlashMessageMiddleware`, and then compose the middleware as a dependency on anything that needs to fetch messages:

```php
public function __construct(FlashMessageMiddleware $flashContainer)
{
    $this->flashContainer = $flashContainer;
}

public function handle(ServerRequestInterface $request): ResponseInterface
{
    $messages = $this->flashContainer->flashMessages($request);
    // ...
}
```

That said, the following would be similar, and can be done _now_:

```php
public function __construct(string $flashAttribute)
{
    $this->flashAttribute = $flashAttribute;
}

public function handle(ServerRequestInterface $request): ResponseInterface
{
    $messages = $request->getAttribute($this->flashAttribute) ?? [];
    // ...
}
```

To sum up:

- Having a configurable attribute name was a design decision to allow segregating messages by domain.
- As a general rule, we do not provide static methods within the framework, and have, in fact, worked to remove them where we previously had them.
- Any helper method would have to be attribute-aware, and provide significant convenience over what is already possible.

**Comment by rieschl** [comment]
Thanks for the detailed explanation! Yes, the configurable attribute key somewhat defeats the usage of such a static method. And of course, coupling in the framework is an issue, Inside my own projects the coupling is less of a problem, so I did not think of that.
But it's okay, I just stick with my own helper method in the project.
